controllers/postgresql/suite_test.go

```diff
@@ -0,0 +1,736 @@
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	err = postgresqlv1alpha1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	//+kubebuilder:scaffold:scheme
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sManager).ToNot(BeNil())
+
+	Expect((&PostgresqlEngineConfigurationReconciler{
+		Client:   k8sClient,
+		Log:      logf.Log.WithName("controllers"),
+		Recorder: k8sManager.GetEventRecorderFor("controller"),
+		Scheme:   scheme.Scheme,
+	}).SetupWithManager(k8sManager)).ToNot(HaveOccurred())
+
+	Expect((&PostgresqlDatabaseReconciler{
+		Client:   k8sClient,
+		Log:      logf.Log.WithName("controllers"),
+		Recorder: k8sManager.GetEventRecorderFor("controller"),
+		Scheme:   scheme.Scheme,
+	}).SetupWithManager(k8sManager)).ToNot(HaveOccurred())
+
+	Expect((&PostgresqlUserReconciler{
+		Client:   k8sClient,
+		Log:      logf.Log.WithName("controllers"),
+		Recorder: k8sManager.GetEventRecorderFor("controller"),
+		Scheme:   scheme.Scheme,
+	}).SetupWithManager(k8sManager)).ToNot(HaveOccurred())
+
+	go func() {
+		defer GinkgoRecover()
+		err = k8sManager.Start(ctx)
+		Expect(err).ToNot(HaveOccurred(), "failed to run manager")
+	}()
+
+	Expect(k8sClient.Create(ctx, &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: pgecNamespace,
+		},
+	})).ToNot(HaveOccurred())
+
+	Expect(k8sClient.Create(ctx, &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: pgdbNamespace,
+		},
+	})).ToNot(HaveOccurred())
+
+	Expect(k8sClient.Create(ctx, &corev1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: pguNamespace,
+		},
+	})).ToNot(HaveOccurred())
+}, 60)
+
+var _ = AfterSuite(func() {
+	cancel()
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+func cleanupFunction() {
+	// Force delete pgec
+	err := deletePGEC(ctx, k8sClient, pgecName, pgecNamespace)
+	Expect(err).ToNot(HaveOccurred())
+	// Force delete secrets
+	err = deleteSecret(ctx, k8sClient, pgecSecretName, pgecNamespace)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(deletePGU(ctx, k8sClient, pguName, pguNamespace)).ToNot(HaveOccurred())
```

Comment:
You must also delete all other names users (master roles, old pgdb name, ...) just in case there is an error on tests